pkg/platform/packet/packet_test.go

```diff
@@ -0,0 +1,169 @@
+package packet
+
+import "testing"
+
+func TestCheckNotEmptyWorkers(t *testing.T) {
+	// Test with empty workers, it should fail.
+	c := config{}
+	hclErr := c.checkNotEmptyWorkers()
+
+	// Exactly 1 failure expected: no worker pools.
+	if len(hclErr) != 1 { // //nolint:gomnd
+		t.Errorf("Expected to detect 1 failure, but detected %v failures", len(hclErr))
+	}
+
+	// Test with one worker pool, it should work.
+	c = config{WorkerPools: []workerPool{{Name: "test"}}}
+	hclErr = c.checkNotEmptyWorkers()
+
+	if hclErr != nil {
+		t.Errorf("Validation failed and shouldn't fail")
+	}
+}
+
+func TestCheckWorkerPoolNamesUniqueDup(t *testing.T) {
+	c := config{
+		WorkerPools: []workerPool{
+			{
+				Name: "dup",
+			},
+			{
+				Name: "dup",
+			},
+		},
+	}
+
+	hclErr := c.checkWorkerPoolNamesUnique()
+
+	// Only one failure expected: the duplicated worker pool.
+	if len(hclErr) != 1 { //nolint:gomnd
+		t.Errorf("Expected to detect 1 failure, but detected %v failures", len(hclErr))
+	}
+}
+
+func TestCheckWorkerPoolNamesUniqueNotDup(t *testing.T) {
+	c := config{
+		WorkerPools: []workerPool{
+			{
+				Name: "not",
+			},
+			{
+				Name: "dup",
+			},
+		},
+	}
+
+	hclErr := c.checkWorkerPoolNamesUnique()
+
+	if hclErr != nil {
+		t.Errorf("Validation failed and shouldn't for worker pool: %v", c.WorkerPools[0].Name)
+	}
+}
+
+func TestCheckResFormatPrefixValidInput(t *testing.T) {
```

Comment:
Tests for `checkResFormat` are not comprehensive. We have 4 execution branches and only 2 tests defined. If you look at the code coverage, it's clearly visible.

Reply:
Thanks to point it out. Tests were not completed, but made sure it includes a test for each error path too now. Thanks :)